proposals/NNNN-developer-oem-communication.md

```diff
@@ -0,0 +1,153 @@
+* Nominal increase in ongoing database storage
+
+## Alternatives Considered
+Several alternatives were considered to store and manage messages and issues for each application instead of a custom solution through SHAID. The primary downsides of most of these alternatives are:
+
+* Requires all Developer Portal users to create accounts on a third-party service
+* Requires all Developer Portal users to link their third-party service account to their Developer Portal account
+* Results in occasional communication by the third-party service without the knowledge/authorization of the SDLC
+* Risk of user terminating their third-party service account
+* Risk of third-party modifying their service offerings or pricing
+
+Brief evaluations of each specific alternative below:
+
+### GitHub
+GitHub offers a comprehensive API to manage repositories, issues, and issue comments, but their documentation is noted with a message that reads:
+
+>The API may change without advance notice during the preview period. Preview features are not supported for production use.
+
+GitHub uses oauth2 for authentication, which would require OEM and app developer users of the Developer Portal to have GitHub accounts and link them to their SDL Developer Portal account. Free private repositories on GitHub only allow 3 collaborators, which would be problematic for apps built by larger teams and/or OEMs with multiple reviewers.
+
+### GitLab
+Similar to GitHub, GitLab also offers a comprehensive oauth2-powered API to manage repositories ("projects"), issues, and issue comments ("issue notes"). There are no limits to the number of private repositories or collaborators, but the friction of requiring all Developer Portal users to register for a GitLab account and link it to the Developer Portal would be a considerable barrier to the user experience.
+
+### Trello
+Trello offers an oauth-based API capable of managing boards, lists, cards, and card comments. There are no documented limits to the number of boards, board members, or lists. Similar to GitHub and GitLab, users would be required to sign up for Trello and link their Trello account to the Developer Portal. Members added to a Trello board via API would likely receive an email from Trello with a direct link to the board, along with other Trello-specific emails.
+
+### Open-source Forum Software
+There are various open-source forum (or forum-like) solutions which we could host such as NodeBB, phpBB, myBB, Discourse, etc, but in my investigation I was only able to find one (Discourse) which offers a comprehensive, documented API. These out-of-the-box generic solutions contain many features which we would likely never use (read: overcomplicated; technical debt), are mostly built upon older technology stacks (PHP, MySQL, Apache) than the rest of SDL's infrastructure, and would likely require heavy customization to achieve seamless integration into the Developer Portal.
```

Comment:
Please refrain from using "I", "we" or "my"

There are various open-source forum (or forum-like) solutions **which **the Developer Portal** could** host such as NodeBB, phpBB, myBB, Discourse, etc, **but when investigating there was** **only one (Discourse)** which offers a comprehensive, documented API. These out-of-the-box generic solutions **contains** many features **which would** likely never  **be used** (read: overcomplicated; technical debt), are mostly built upon older technology stacks (PHP, MySQL, Apache) than the rest of SDL's infrastructure, and would likely require heavy customization to achieve seamless integration into the Developer Portal.